### ISSUE TITLE
Tool to represent yaml model to graphical representation!

### ISSUE THREAD

**Issue by PraveenMNaik** [ultralytics/yolov5#8845]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Is there any tool available to represent any .yaml fiile , representing model to graphical or model representation.

**Comment by AyushExel** [comment]
@PraveenMNaik I like the idea of model visualizer too but I think there's a problem in creating such tool. The model definitions and structure need to be parsed and represented and there isn't a standard to do both. Every framework has its own way of representing/parsing yaml. If you come across any such tool, please let me know..

Meanwhile, yolov5 provides a way to visualize the model in cli via yolo.py module
```
python models/yolo.py --cfg yolov5s.yaml
```

**Comment by PraveenMNaik** [comment]
Thank you for the reply @AyushExel 

I run the as said....But i didn't understand the below...
**YOLOv5s summary: 270 layers, 7235389 parameters, 7235389 gradients, 16.6 GFLOPs

Fusing layers... 
YOLOv5s summary: 213 layers, 7225885 parameters, 7225885 gradients, 16.4 GFLOPs**

whats the difference between normal summary and fusing layers summary....

**Comment by Zephyr69** [comment]
I don't think that exists; it has to be written specifically for yolov5.

The cleanest presentation I've seen so far is to save a model, convert to .onnx, and view it in Netron.

**Comment by glenn-jocher** [comment]
@PraveenMNaik 👋 Hello! Thanks for asking about YOLOv5 🚀 **architecture visualization**. Fusing Conv+BatchNorm layers will reduce layer and parameter counts.

We've made visualizing YOLOv5 🚀 architectures super easy. There are 3 main ways:

## `model.yaml`

Each model has a corresponding yaml file that displays the model architecture. Here is YOLOv5s, defined by `yolov5s.yaml`:
https://github.com/ultralytics/yolov5/blob/1a3ecb8b386115fd22129eaf0760157b161efac7/models/yolov5s.yaml#L12-L48

## TensorBoard Graph

Simply start training a model, and then view the TensorBoard Graph for an interactive view of the model architecture. This example shows YOLOv5s viewed in our [Notebook](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb) –  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

```python
# Tensorboard
%load_ext tensorboard
%tensorboard --logdir runs/train

# Train YOLOv5s on COCO128 for 3 epochs
python train.py --weights yolov5s.pt --epochs 3
```

<img width="893" alt="Screenshot 2021-04-11 at 01 10 09" src="https://user-images.githubusercontent.com/26833433/114286928-349bd600-9a63-11eb-941f-7139ee6cd602.png">

## Netron viewer

Use [https://netron.app](https://netron.app/) to view exported ONNX models:

```
python export.py --weights yolov5s.pt --include onnx --simplify
```

<img width="890" alt="Screen Shot 2022-04-29 at 11 09 23 AM" src="https://user-images.githubusercontent.com/26833433/165999628-0d6095ce-58dd-49b7-a04e-25d0995e5493.png">


Good luck 🍀 and let us know if you have any other questions!

**Comment by PraveenMNaik** [comment]
Thank you Sir.

**Comment by glenn-jocher** [comment]
@PraveenMNaik you're welcome! If you have any more questions or need further assistance, feel free to ask. Good luck with your YOLOv5 project! 🚀
